C++/test/data_structure/tree/binary_search_tree.cpp

```diff
@@ -0,0 +1,40 @@
+#include "third_party/catch.hpp"
+#include "data_structure/tree/binary_search_tree.hpp"
+
+TEST_CASE("Create an empty search tree", "[binary-search-tree]") {
+	BinarySearchTree tree;
+	REQUIRE(tree.traversal_inorder_recursive() == vector<int>({})); 	// empty vector
+	REQUIRE(tree.traversal_inorder_iterative() == vector<int>({})); 	// empty vector
+	REQUIRE(tree.traversal_preorder_recursive() == vector<int>({})); 	// empty vector
+	REQUIRE(tree.traversal_preorder_iterative() == vector<int>({})); 	// empty vector
+	REQUIRE(tree.traversal_postorder_recursive() == vector<int>({})); 	// empty vector
+	REQUIRE(tree.traversal_postorder_iterative() == vector<int>({})); 	// empty vector
+}
+
+TEST_CASE("Add and remove values", "[binary-search-tree]") {
+	BinarySearchTree tree;
+	tree.insert(1);
+	tree.insert(2);
+	REQUIRE(tree.search(1) == true);
+	tree.remove(1);
+	REQUIRE(tree.search(1) == false);
+	REQUIRE(tree.search(2) == true);
```

Comment:
Having more insertions, removals and searches here would be good as that would make the test more robust, and therefore, more reliable.

Reply:
Good thing you asked for this, because it turns out that removing three elements in a row throws a SIGABRT. I'll run that down and add it to a new commit.

Reply:
Looks like `remove()` was missing a return statement. Added a bunch of tests and they're all passing. Thanks again for the nudge.